istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	admit_v1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/api/label"
+)
+
+const (
+	// TODO(monkeyanator) move into istio/api
+	istioTagLabel             = "istio.io/tag"
```

Comment:
does this belong in istio/api repo if users aren't meant to directly set it?

Reply:
Not for now, no. Once the feature gets more mature users might want to use it as a selector or something so it might make sense to put it there eventually.

Reply:
https://github.com/istio/istio/pull/29713 - I too am using it in my PR. That is for giving revision centric view which also includes tags also lists all tags pointing to a revision. It would be better to move to a common place and `istio/api` labels package seems to be the right place for this one.

Reply:
Sorry, I might have misunderstood the question. Yes, if it belongs any common place then istio/api is the right place for it. However, I don't think it's necessary to block this PR on getting it into istio/api.

Reply:
Yeah. I agree. There is no need to block this PR. But eventually, we have to clean this up.